cxplat/inc/cxplat/memory.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    /**
+     * @brief Allocate memory.
+     * @param[in] size Size of memory to allocate.
+     * @returns Pointer to memory block allocated, or null on failure.
+     */
+    _Must_inspect_result_ _Ret_writes_maybenull_(size) void* cxplat_allocate(size_t size);
```

Comment:
Super happy to see the SAL annotations, but you will need a stub file for posix platforms. Feel free to steal [quic_sal_stub.h](https://github.com/microsoft/msquic/blob/main/src/inc/quic_sal_stub.h) and rename to `cxplat_sal_stub.h` or whatever you'd like.

Reply:
agreed.  not needed for this PR but agree it will be needed eventually.

Reply:
I propose we do this after moving code to the cxplat repo, since a stub file isn't needed for the usersim project which does not support posix platforms.